test/errors.js

```diff
@@ -32,8 +31,14 @@ function generateDisconnectionCase(readyKillIdFun, cleanupKillIdFun) {
 
     function killThread(argFun) {
       var threadId = argFun(zongji);
-      test.ok(!isNaN(threadId));
-      conn.db.query('KILL ' + threadId);
+      if (typeof threadId === 'object') {
+        threadId.end(function (err) {
+          test.done();
```

Comment:
How does this work correctly when the `ctrlPool` is killed first? Seems like it would call the `test.done()` before finishing the test.